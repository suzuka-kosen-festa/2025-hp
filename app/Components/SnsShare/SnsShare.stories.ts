import type { Meta, StoryObj } from "@storybook/react";
import { SnsShare } from "./SnsShare";

const meta: Meta<typeof SnsShare> = {
  title: "Components/SnsShare",
  component: SnsShare,
};

export default meta;
type Story = StoryObj<typeof SnsShare>;

export const Default: Story = {
  args: {},
};
