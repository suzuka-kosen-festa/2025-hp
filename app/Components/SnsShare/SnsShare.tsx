import type { FC } from "react";

export const SnsShare: FC = () => {
  const url = "https://kosenfes-2025.trap.show";
  const text = "高専祭2025 #高専祭2025";

  return (
    <div>
      <a
        href={`https://twitter.com/intent/tweet?url=${url}&text=${text}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        X
      </a>
      <a
        href={`https://www.facebook.com/sharer/sharer.php?u=${url}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        Facebook
      </a>
      <a
        href={`https://line.me/R/msg/text/?${text}%0A${url}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        LINE
      </a>
    </div>
  );
};
