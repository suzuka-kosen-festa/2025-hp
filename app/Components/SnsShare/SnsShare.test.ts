// @vitest-environment jsdom

import { composeStories } from "@storybook/react";

import { expect, it } from "vitest";

import * as stories from "./SnsShare.stories";

const { Default } = composeStories(stories);
it("snsShare snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
