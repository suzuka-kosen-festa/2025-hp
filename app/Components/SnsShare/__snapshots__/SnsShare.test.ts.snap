// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SnsShare snapshot 1`] = `
<div>
  <div>
    <a
      href="https://twitter.com/intent/tweet?url=https://kosenfes-2025.trap.show&text=高専祭2025 #高専祭2025"
      rel="noopener noreferrer"
      target="_blank"
    >
      X
    </a>
    <a
      href="https://www.facebook.com/sharer/sharer.php?u=https://kosenfes-2025.trap.show"
      rel="noopener noreferrer"
      target="_blank"
    >
      Facebook
    </a>
    <a
      href="https://line.me/R/msg/text/?高専祭2025 #高専祭2025%0Ahttps://kosenfes-2025.trap.show"
      rel="noopener noreferrer"
      target="_blank"
    >
      LINE
    </a>
  </div>
</div>
`;
